test/utils/audit.go

```diff
@@ -101,6 +102,9 @@ func parseAuditLine(line string, version schema.GroupVersion) (AuditEvent, error
 	if e.RequestObject != nil {
 		event.RequestObject = true
 	}
+	if e.ImpersonatedUser != nil {
+		event.ImpersonatedUser = e.ImpersonatedUser.Username
+	}
```

Comment:
Do you think we should verify impersonated groups too?

Reply:
Yeah, I think thats a good idea! I extended the tests to also check the groups.
However, as you can see some (seemingly) unrelated tests are failing now.  I ran the tests twice and both times the tests that are failing are different, while `Advanced Audit should audit API calls ` passes both times.